### PR TITLE
Add support for a `val` split

### PR DESCRIPTION
**Pull request by damian0815**
Separate validation split and support for stabilising the loss graph as training progresses.

Improves the display of losses as training progresses from this:
basic-losses

to this:
validation-losses

By looking at the new `loss/val` graph on the right side of the second image, you can see that the model is clearly not learning much new things after around step 60 or 70, and it is clearly starting to fry at around step 120.

### Code changes

* DataLoaderMultiAspect is now non-singleton and supports removing items and optional init from a list of ImageTrainItem instead of a root folder path
* Multiple EveryDreamBatch instances can be created each with their own DataLoaderMultiAspect
* Added "validation_config" entry to config.json that points to a separate validation_config.json file (example in validation_default.json)
* Validation split support, either automatic by stealing items from the `train` dataset or manual by providing a separate `val` data root folder with captioned images to use for validation
* Validation config also has an option to stabilise the `train` loss graph by performing an additional loss calculation pass over an overlapping subset of the training data with a fixed seed

### CODE REVIEW

**Comment by damian0815** [comment]
closing in favour of @noprompt 's #50
